jwst/extract_1d/extract.py

```diff
@@ -1316,8 +1316,9 @@ def do_extract1d(input_model, refname, smoothing_length, bkg_order):
                     spec.name = slitname
                 output_model.spec.append(spec)
 
-        elif isinstance(input_model, datamodels.CubeModel):
+        elif isinstance(input_model, (datamodels.CubeModel, datamodels.SlitModel)):
```

Comment:
So do the NIRSpec BrightObj exposures now come through here as a SlitModel, instead of CubeModel? If so, I guess that implies that SlitModel can handle 3D data?

Reply:
Yes, NRS_BRIGTOBJ comes out of extarct_2d as a SlitModel. SlitModel acts as enhanced version of either ImageModel or CubeModel. The issue with the test is solved in the SlitModel class. When the input to a  SlitModel is a SlitModel (instead of a file) it was not copying the wavelength array and that forced it to created a wavelength array with the sahpe of the data which is 3D. For NRS_BRIGHTOBJ the dat ais 3D but the wavelength array is 2D.

Reply:
Oh, this also means that the output of calweb_spec2 is a SlitModel in this case.
Does this have any implications for the spec3 pipeline?

Reply:
Possibly. We'll obviously need to check that. calwebb_spec3 might also need updates to account for the fact that input IFU files are no longer an ImageModel, but IFUImageModel. As I looked at the changes in this PR it also made me wonder if the Photom step is now going to have issues because of the NIRSpec BrightObj data coming through as SlitModel, instead of CubeModel. I guess the reg tests will let us know.

Reply:
The photom step was also adjusted in this PR.